### PR TITLE
Redo app design page to comply with updated study object.

### PR DESCRIPTION
**Pull request by leonli33**

### CODE REVIEW

File: src/components/studies/StudyBuilder.tsx [hunk 1]
```diff
@@ -162,14 +157,17 @@ const StudyBuilder: FunctionComponent<StudyBuilderProps> = ({
...
+
     try {
       const newVersion = await StudyService.updateStudy(study, token!)
       const updatedStudy = { ...study, version: newVersion }
       setData({
         ...builderInfo,
         study: updatedStudy,
       })
-
+      if (error !== '') {
```
Comment (conundrumgirl) [inline src/components/studies/StudyBuilder.tsx:168]: do we need this check?

File: src/components/studies/StudyBuilder.tsx [hunk 2]
```diff
@@ -391,24 +390,19 @@ const StudyBuilder: FunctionComponent<StudyBuilderProps> = ({
...
                         saveLoader={saveLoader}
                         id={id}
-                        currentAppDesign={
-                          builderInfo.study.clientData?.appDesign ||
-                          ({} as StudyAppDesign)
-                        }
-                        onSave={() => {
+                        study={builderInfo.study}
+                        onSave={(updatedStudy: Study) => {
+                          setData({
```
Comment (conundrumgirl) [inline src/components/studies/StudyBuilder.tsx:395]: saveStudy does setData. Do we need to call setData again?

**Review by conundrumgirl — APPROVED** [review]

File: src/components/studies/StudyBuilder.tsx (hunk 1 above)
Comment (leonli33) [reply src/components/studies/StudyBuilder.tsx:168]: Yes, if we do not do this then the error message will still show on top of the screen even if the study was updated successfully.

File: src/components/studies/StudyBuilder.tsx (hunk 2 above)
Comment (leonli33) [reply src/components/studies/StudyBuilder.tsx]: Updated.

File: src/components/studies/StudyBuilder.tsx (hunk 1 above)
Comment (conundrumgirl) [reply src/components/studies/StudyBuilder.tsx:168]: What I'm questioning is the needs to check.  I think just doing setError('') is sufficient. Yes it's extra set if it's already empty but it removes the check an decreases mental load for a person who might need to look at the code later. It's really a tiny detail.

File: src/components/studies/app-design/AppDesign.tsx [hunk 3]
```diff
@@ -610,9 +610,9 @@ const AppDesign: React.FunctionComponent<
       const formattedPhone = formatPhoneNumber(phoneWithoutZipcode)
       setIrbPhoneNumber(formattedPhone)
     }
-    const isWelcomeScreenDataIsEmpty =
+    const isWelcomeScreenDataEmpty =
       Object.keys(study.clientData.welcomeScreenData || {}).length == 0
-    const welcomeScreenData = isWelcomeScreenDataIsEmpty
+    const welcomeScreenData = isWelcomeScreenDataEmpty
```
Comment (conundrumgirl) [inline src/components/studies/app-design/AppDesign.tsx:615]: can it just be `welcomeScreenData  =  Object.keys(study.clientData.welcomeScreenData || {}).length == 0`

**Pull request merged**